ossdbtoolsservice/query/batch.py

```diff
@@ -163,6 +163,10 @@ def save_as(self, params: SaveResultsRequestParams, file_factory: FileStreamFact
 
         self._result_set.save_as(params, file_factory, on_success, on_failure)
 
+    def notice_handler(self, notice: str, conn: ServerConnection):
+        if not (conn.user_transaction and notice.message_primary == 'there is already a transaction in progress'):
```

Comment:
Is there any specific reason why we add the message when the message is 'there is already a transaction in progress'):?

Reply:
Added updates to this:
It seems by default Psycopg3 is always starting a transaction. So when executing "begin" command, it sends warning that one has already started. I added a check so we only show this warning when a transaction is being started while still currently in a user started transaction block.
<img width="435" alt="image" src="https://github.com/microsoft/pgtoolsservice/assets/69922333/54a62e3f-a21b-42c6-9930-0a60fddd65ea">